### PR TITLE
feat: streetname name was corrected

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/StreetNameRegistry/Municipality/Commands/CorrectStreetNameName.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    using Be.Vlaanderen.Basisregisters.Utilities;
+
+    public class CorrectStreetNameName : IHasCommandProvenance
+    {
+        private static readonly Guid Namespace = new Guid("65740d14-9ef4-49e6-8c8b-3a4108bcd831");
+
+        public MunicipalityId MunicipalityId { get; }
+        public Provenance Provenance { get; }
+        public Names StreetNameNames { get; }
+        public PersistentLocalId PersistentLocalId { get; }
```
Comment (ArneD) [inline src/StreetNameRegistry/Municipality/Commands/CorrectStreetNameName.cs:16]: Please move as second property, also in ctor.

File: src/StreetNameRegistry/Municipality/Events/StreetNameNameWasCorrectedV2.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        public const string EventName = "StreetNameNameWasCorrectedV2"; // BE CAREFUL CHANGING THIS!!
+
+        [EventPropertyDescription("Interne GUID van de gemeente aan dewelke de straatnaam is gekoppeld.")]
+        public Guid MunicipalityId { get; }
+
+        [EventPropertyDescription("De straatnamen in de officiële en (eventuele) faciliteitentaal van de gemeente. Mogelijkheden: Dutch, French, German of English.")]
+        public List<StreetNameName> StreetNameNames { get; }
+
+        [EventPropertyDescription("Objectidentificator van de straatnaam.")]
+        public int PersistentLocalId { get; }
```
Comment (ArneD) [inline src/StreetNameRegistry/Municipality/Events/StreetNameNameWasCorrectedV2.cs:26]: Please move as second property, also in ctors.

File: test/StreetNameRegistry.Tests/AggregateTests/WhenCorrectingStreetNameName/GivenMunicipality.cs [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+                    new Names(new []
+                    {
+                        new StreetNameName("Kapelstraat", Language.Dutch),
+                        new StreetNameName("Rue de la Croix - Rouge", Language.French)
+                    }),
+                    command.PersistentLocalId))));
+        }
+
+        [Fact]
+        public void ThenStreetNameNotFoundExceptionWasThrown()
```
Comment (ArneD) [inline test/StreetNameRegistry.Tests/AggregateTests/WhenCorrectingStreetNameName/GivenMunicipality.cs:66]: Here you use `xxxWasThrown` and a few below as well. But beneath them there are `_ThenThrowxxx`.
Could you make it consistent and use `ThenThrows`?

**Review by ArneD — CHANGES_REQUESTED** [review]
Also include projection tests.

**Review by pgallik — CHANGES_REQUESTED** [review]

File: src/StreetNameRegistry.Api.BackOffice.Abstractions/Extensions/BackOfficeContextExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    public static class BackOfficeContextExtensions
+    {
+        public static async Task<MunicipalityId> GetMunicipalityIdByPersistentLocalId(
+            this BackOfficeContext backOfficeContext,
+            int streetNamePersistentLocalId)
+        {
+            var municipalityIdByPersistentLocalId = await backOfficeContext
+                .MunicipalityIdByPersistentLocalId
+                .FindAsync(streetNamePersistentLocalId);
+
```
[the remark this answers is not in the document]
Comment (emalfroy) [reply src/StreetNameRegistry.Api.BackOffice.Abstractions/Extensions/BackOfficeContextExtensions.cs:21]: done

File: src/StreetNameRegistry/Municipality/Events/StreetNameNameWasCorrectedV2.cs (hunk 2 above)
Comment (emalfroy) [reply src/StreetNameRegistry/Municipality/Events/StreetNameNameWasCorrectedV2.cs]: done

File: src/StreetNameRegistry/Municipality/Municipality.cs [hunk 5]
```diff
@@ -98,6 +98,39 @@ public void RetireStreetName(PersistentLocalId persistentLocalId)
             streetName.Retire();
         }
 
+        public void CorrectStreetNameName(Names streetNameNames, PersistentLocalId persistentLocalId)
+        {
+            if (!StreetNames.HasPersistentLocalId(persistentLocalId, out var streetName) || streetName is null)
```
[the remark this answers is not in the document]
Comment (emalfroy) [reply src/StreetNameRegistry/Municipality/Municipality.cs]: done

**Review by ArneD — APPROVED** [review]

**Pull request merged**